llvm/tools/sycl-post-link/ModuleSplitter.cpp

```diff
@@ -562,6 +562,21 @@ void ModuleDesc::cleanup() {
   MPM.run(*M, MAM);
 }
 
+bool ModuleDesc::isSpecConstantDefault() const {
+  return Props.IsSpecConstantDefault;
+}
+
+void ModuleDesc::setSpecConstantDefault(bool Value) {
+  Props.IsSpecConstantDefault = Value;
+}
+
+ModuleDesc CloneModuleDesc(const ModuleDesc &MD) {
+  std::unique_ptr<Module> NewModule = CloneModule(MD.getModule());
+  ModuleDesc NewMD(std::move(NewModule));
+  NewMD.EntryPoints.Props = MD.EntryPoints.Props;
+  return NewMD;
+}
```

Comment:
It seems like some kind of deep copy of ModuleDesc is needed here. At least in addition to EntryPoints.Props also EntryPoints.Functions need to be copied.
Otherwise cloned module/device image is not registered in ProgramManager through void ProgramManager::addImages(pi_device_binaries DeviceBinary)  because it doesn't have entry points.

Reply:
Addresses it. Now symbol generation is being checked in `llvm/test/tools/sycl-post-link/spec-constants/default-value/device-image.ll`